data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/configuration/S3ScanScanOptions.java

```diff
@@ -40,6 +40,9 @@ public class S3ScanScanOptions {
     @JsonProperty("end_time")
     private LocalDateTime endTime;
 
+    @JsonProperty("metadata_only")
```

Comment:
I think we should make this more generic by use of an enum. There may be different configurations that users want:

1. Object data only (what we have now)
2. Object data and object metadata
3. Metadata only

Also, this option should exist on the high level configuration for `s3`. It is applicable for both SQS and S3 scan.

Reply:
We do not have "object data only" we already have object data and metadata (bucket and key) but not complete metadata. So, not sure how we want to handle that

Reply:
Also "metadata only" really doesn't make sense for s3-sqs when we have sqs source. I guess we could still do it.